domain/src/main/java/org/oppia/android/domain/survey/SurveyConstantsProvider.kt

```diff
@@ -0,0 +1,17 @@
+package org.oppia.android.domain.survey
+
+import dagger.Module
+import dagger.Provides
+import javax.inject.Qualifier
+
+/** Qualifier corresponding to the total number of questions in a survey. */
+@Qualifier
+annotation class TotalQuestionCount
+
+/** Provider to return the constants required during the survey. */
+@Module
+class SurveyQuestionModule {
+  @Provides
+  @TotalQuestionCount
+  fun provideTotalQuestionCount(): Int = 4
```

Comment:
Just wondering -- isn't the list of questions stored somewhere else, and should this be computed from that list rather than be declared directly? It seems like a separate source of truth and I'm not sure what would happen if someone edited the list of questions but not this constant, or vice versa.

Reply:
We generate the list at the beginning of the survey, and do not store it. It only exists ephemerally during the survey.
We have an enum class of all the question names needed in the survey, and from this, I generate the actual question objects. The challenge I had is that this enum includes 6 questions, where the last 3 are conditional based on the answer selected in the 3rd question(nps score). So I opted to take the first four questions in the list, which is what this value  is used for, and then switch out the last question dynamically depending on the nps score selected.

If someone were to add a new question to the list and not edit the constant, then that could actually create a problem based on the position of the question, since the list is indexed.

I think I will need to fix this, but I am wondering whether what would be the best way to dictate the number of questions that go in the survey - a platform parameter or maybe just a variable in the controller. More drastically, the question list generation logic could change, but let me explore the former option first.

Reply:
Following our conversation to avoid  using two sources of data that are dependent but represented independently, I came up with a solution that is more in line with your comment here:

It is now the responsibility of the caller of the  `startSurveySession()` to provide the list of questions at call site, and we compute the number of questions from this. I have overall modified how the survey is generated to support this, but the rest of the architecture remains intact.